Lib/configparser.py

```diff
@@ -773,11 +776,13 @@ def read_file(self, f, source=None):
             except AttributeError:
                 source = '<???>'
         self._read(f, source)
+        self._loaded_sources.append(source)
 
     def read_string(self, string, source='<string>'):
         """Read configuration from a given string."""
         sfile = io.StringIO(string)
         self.read_file(sfile, source)
+        self._loaded_sources.append(source)
```

Comment:
This will be duplicated because `read_file` already does `self._loaded_sources.append(source)`.

Reply:
Yes, I have added this check inside unittest as well.